packages/api/test/e2e/doughnut.e2e.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright 2019 Centrality Investments Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import testingPairs from '@plugnet/keyring/testingPairs';
+import {waitReady} from '@polkadot/wasm-crypto';
+import {Api} from '../../src/Api';
+import {Doughnut} from '@cennznet/types/extrinsic/Doughnut';
+import {KeyringPair} from '@cennznet/util/types';
+import {Keypair} from '@plugnet/util-crypto/types';
+import {hexToU8a} from '@plugnet/util'
+
+const doughnut_maker = require('doughnut-maker');
+const cennznut = require('cennznut.js');
+
+/**
+ * Create a new unsigned Doughnut
+ */
+function makeDoughnut(holder: KeyringPair, issuer: Keypair): Doughnut {
+
+  // Create a valid CENNZnet domain permissions
+  const cennznetDomain = {
+    modules: {
+      'generic-asset': {
+        methods: {
+          'transfer': {
+            blockCooldown: 100
+          }
+        }
+      }
+    }
+  };
+
+  // Hack together an encoded domain for now
+  const encodedDomainPayload = cennznut.versions['0'].encode(cennznetDomain);
+  const encodedDomain = new Uint8Array(encodedDomainPayload.length + 2);
+  encodedDomain.set(encodedDomainPayload, 2);
+
+  const expiry = Math.ceil(Date.now() / 1000) + 60;
+
+  return new Doughnut(doughnut_maker.generate(0, 0, {
```

Comment:
Will it be better if cennznut can integrate doughutmaker and expose generate(), hiding encoding details in itself.